### DISCUSSION TITLE
The set_pipeline step should unpause pipelines it archived when unarchiving them, or include a flag to do so

### DISCUSSION THREAD

**Discussion by Clockwork757** [concourse/concourse#8135] [Ideas]
We manage a few hundred generated pipelines that get set from one job. Recently due to an upstream change in our source of truth one of the generated pipelines was deleted, which archived it. After it was restored we didn't notice that it was paused for a while. We expected it to be unpaused like any other pipeline created via `set_pipeline`. I can't find this specific behavior documented anywhere.

I think it would make sense for `set_pipeline` to just unarchive pipelines to an unpaused state, since it's no different from creating a pipeline that has some history. I would also be okay with adding a flag on the step to enable this behavior.

**Comment by saj** [comment]
Yes, please!  I was very surprised to learn that `set_pipeline` is not deterministic.

In our case, we `set_pipeline` on the appearance of a new VCS branch.  With current Concourse behaviour:

1. A new branch head with name `foo` appears.  `set_pipeline` creates a new (unpaused) pipeline and all is hunky dory.
2. Branch `foo` is unpushed.  The pipeline is archived.  OK.
3. A new branch head with name `foo` appears.  `set_pipeline` does nothing.
